test/unit/KNXLog.test.ts

```diff
@@ -0,0 +1,67 @@
+import { describe, it, beforeEach, afterEach } from 'node:test'
+import assert from 'node:assert'
+import { module as createLogger, setLogLevel } from '../../src/KnxLog'
+import winston from 'winston'
+
+describe('KNX Logger Functionality Tests', async () => {
+	let originalEnv: NodeJS.ProcessEnv
+
+	beforeEach(() => {
+		originalEnv = { ...process.env }
+	})
+
+	afterEach(() => {
+		process.env = originalEnv
+	})
+
+	it('should create logger with correct module name', () => {
+		const logger = createLogger('TestLogger')
+		assert.strictEqual(logger.module, 'TestLogger')
+	})
+
+	it('should set log level correctly', () => {
+		const logger = createLogger('TestLogger')
+		setLogLevel('error')
+		assert.strictEqual(logger.level, 'error')
+	})
+
+	it('should apply log level to all loggers', () => {
+		const logger1 = createLogger('Logger1')
+		const logger2 = createLogger('Logger2')
+
+		setLogLevel('debug')
+
+		assert.strictEqual(logger1.level, 'debug')
```

Comment:
Add a second call to setLogLevel and ensure the log level is changed again, this prevents cases where you are comparing to a default logelevel